src/ucp/proto/proto_select.inl

```diff
@@ -120,31 +120,57 @@ ucp_proto_select_lookup(ucp_worker_h worker, ucp_proto_select_t *proto_select,
 /*
  * @note op_attr_mask is from @ref ucp_request_param_t, defined by @ref ucp_op_attr_t.
  */
-static UCS_F_ALWAYS_INLINE void
-ucp_proto_select_param_init(ucp_proto_select_param_t *select_param,
-                            ucp_operation_id_t op_id, uint32_t op_attr_mask,
-                            unsigned op_flags, ucp_dt_class_t dt_class,
-                            const ucp_memory_info_t *mem_info, uint8_t sg_count)
+static UCS_F_ALWAYS_INLINE void ucp_proto_select_param_init_common(
+        ucp_proto_select_param_t *select_param, ucp_operation_id_t op_id,
+        uint32_t op_attr_mask, unsigned op_flags, ucp_dt_class_t dt_class,
+        const ucp_memory_info_t *mem_info, uint8_t sg_count)
 {
     if (dt_class == UCP_DATATYPE_CONTIG) {
         ucs_assert(sg_count == 1);
     } else if (dt_class != UCP_DATATYPE_IOV) {
         ucs_assert(sg_count == 0);
     }
 
+    ucs_assertv(!(op_id & op_flags), "op_id=0x%x op_flags=0x%x", op_id,
+                op_flags);
+    UCS_STATIC_ASSERT(sizeof(select_param->op.padding) ==
+                      sizeof(select_param->op));
+
     /* construct a protocol lookup key based on all operation parameters
      * op_flags are modifiers for the operation, for now only FAST_CMPL is
      * supported */
-    ucs_assertv(!(op_id & op_flags), "op_id=0x%x op_flags=0x%x", op_id,
-                op_flags);
-    select_param->op_id_flags = op_id | op_flags;
-    select_param->op_attr     = ucp_proto_select_op_attr_pack(op_attr_mask);
-    select_param->dt_class    = dt_class;
-    select_param->mem_type    = mem_info->type;
-    select_param->sys_dev     = mem_info->sys_dev;
-    select_param->sg_count    = sg_count;
-    select_param->padding[0]  = 0;
-    select_param->padding[1]  = 0;
+    select_param->op_id_flags   = op_id | op_flags;
+    select_param->op_attr       = ucp_proto_select_op_attr_pack(op_attr_mask);
+    select_param->dt_class      = dt_class;
+    select_param->mem_type      = mem_info->type;
+    select_param->sys_dev       = mem_info->sys_dev;
+    select_param->sg_count      = sg_count;
+    select_param->op.padding[0] = 0;
+    select_param->op.padding[1] = 0;
+}
+
+static UCS_F_ALWAYS_INLINE void
+ucp_proto_select_param_init(ucp_proto_select_param_t *select_param,
+                            ucp_operation_id_t op_id, uint32_t op_attr_mask,
+                            unsigned op_flags, ucp_dt_class_t dt_class,
+                            const ucp_memory_info_t *mem_info, uint8_t sg_count)
+{
+    ucp_proto_select_param_init_common(select_param, op_id, op_attr_mask,
+                                       op_flags, dt_class, mem_info, sg_count);
+    select_param->op.padding[0] = 0;
+    select_param->op.padding[1] = 0;
```

Comment:
The padding has been already zeroed in `ucp_proto_select_param_init_common`.
The extra function can be removed, or these lines can be removed from the common function.